arrow-docs/docs/docs/optics/filterindex/README.MD

```diff
@@ -0,0 +1,51 @@
+---
+layout: docs
+title: FilterIndex
+permalink: /docs/optics/filterindex/
+---
+
+## FilterIndex
+
+`FilterIndex` provides a [Traversal]({{ '/docs/optics/traversal' | relative_url }}) that can see into a structure `S` and get, set or modify 0 to N foci whose index `I` satisfies a predicate.
+
+If for a structure `S` the foci `A` can be indexed by `I` then a `Traversal` can be created by `FilterIndex` that is filtered by a predicate on `I`.
+
+`FilterIndex` can easily be created given a `Traverse` instance and an indexing function.
+
+```kotlin:ank
+import arrow.core.*
+import arrow.data.*
+import arrow.optics.*
+import arrow.optics.typeclasses.*
+
+val filterIndexStringByIndex = FilterIndex.fromTraverse<ListKWHK, String>({ list ->
+    list.ev().map {
+        it toT it.length
+    }
+}, ListKW.traverse())
+```
+
+Given a `FilterIndex` instance we can create a `Traversal` that filters out the foci that do not match the predicate.
+
+```kotlin:ank
+val filter: Traversal<ListKWKind<String>, String> = filterIndexStringByIndex.filter { length -> length > 3 }
+
+filter.getAll(listOf("H", "He", "Hel", "Hell", "Hello").k())
+```
+
+Arrow provides some instances like `ListKW` indexed by index position.
```

Comment:
I'm not sure what this means, and I'd reference the call required to get these instances in this paragraph so people understand which part of the snippet you're referring to.